src/commands/dev/lib/yarn-installer.ts

```diff
@@ -0,0 +1,71 @@
+import execa from 'execa';
+import { tmpdir } from 'os';
+import {
+  pathExists,
+  mkdirp,
+  createWriteStream,
+  statSync,
+  chmodSync
+} from 'fs-extra';
+import pipe from 'promisepipe';
+import { join } from 'path';
+import fetch from 'node-fetch';
+import tar from 'tar-fs';
+import { setScriptGlobals } from '@now/build-utils';
+import { Output } from '../../../util/output/create-output';
+
+async function isFoundInPath(): Promise<boolean> {
+  const [npm, yarn] = await Promise.all([
+    execa('npm', ['--version']),
+    execa('yarn', ['--version'])
+  ]);
+  return typeof npm === 'string' || typeof yarn === 'string';
+}
+
+async function installYarn(output: Output): Promise<string> {
+  // Loosely based on https://yarnpkg.com/install.sh
+  const dirName = join(tmpdir(), 'co.zeit.now', 'dev', 'yarn');
+  // const full = join(dirName, 'yarn-latest.tar.gz');
+  const yarnBin = join(dirName, 'bin', 'yarn');
+
+  if (await pathExists(yarnBin)) {
+    output.debug('The yarn module is already cached, not re-downloading');
+    return yarnBin;
+  }
+
+  output.debug(`Creating directory ${dirName}`);
+  await mkdirp(dirName);
+  output.debug(`Finished creating ${dirName}`);
+
+  const url = 'https://yarnpkg.com/latest.tar.gz';
```

Comment:
I think we should hard-code the release version, so that we can include the expected shasum in this file. Perhaps https://github.com/yarnpkg/yarn/releases/tag/v1.15.2

Reply:
Also just downloading this JS file seems easier to me than all the tar extraction stuff https://github.com/yarnpkg/yarn/releases/download/v1.15.2/yarn-1.15.2.js

Reply:
Done!